### PR TITLE
feature: eth-wire types standalone crate

### LINKED ISSUE #6959: Move eth-wire types to a standalone crate
### Describe the feature

currently reth-interfaces is pretty big and depends on things like:

https://github.com/paradigmxyz/reth/blob/53e67c2b99c0dba0b9636c1e94d3184661257b50/crates/interfaces/Cargo.toml#L18-L19

I think we can solve this by separating the wire types from the implementation by moving them to their own crate: eth-wire-types

### Additional context

This is currently blocked until audit concluded

_No response_

### PR DESCRIPTION
**Pull request by JackG-eth**
close #6959 

move eth-wire types to its own standalone crate to reduce the size of reth-interface crate.

### CODE REVIEW

File: crates/net/eth-wire-types/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//! Types for the eth wire protocol.
```
Comment (mattsse) [inline crates/net/eth-wire-types/src/lib.rs:1]: this needs the regular header

like this:

https://github.com/paradigmxyz/reth/blob/33c466d88c4bee67f0b4eb15b2ca1b2fc421b10b/crates/net/nat/src/lib.rs#L7-L12

**Review by mattsse — CHANGES_REQUESTED** [review]
great start

**Comment by JackG-eth** [comment]
@mattsse I think it can be reviewed, will clean up linting/doc issues.

1 thing, maybe my lack of knowledge but i can swap the interface dependency to types due to optimism = ["reth-eth-wire/optimism"]

**Review by mattsse — APPROVED** [review]
lgtm

pending @Rjected

**Review by Rjected — APPROVED** [review]
looks good to me, thanks!

**Pull request merged**